Update design - use Sveltekit

Visual-Dawg opened the pull request:
Use Sveltekit to generate the site.
Also worked a bit on the design and stuff

LeoDog896 commented:
Hello!

The author isn't doing any PRs that introduce "[unnecessary refactors](https://github.com/hyprwm/hyprland-website/issues/17#issuecomment-1412248460)" as the author doesn't want to use frameworks in their website (See https://github.com/hyprwm/hyprland-website/issues/17) in order to avoid [bikeshedding](https://en.wikipedia.org/wiki/Law_of_triviality) and arguments, nor are they accepting any deployment strategy changes (I noticed you're using [@sveltejs/adapter-static](https://github.com/hyprwm/hyprland-website/pull/28/files#diff-6317e218ea40a2a9b47af98e100e81c06f9c0abff66737e6ca06ec4b29402242), but the author also doesn't want any of these changes either (https://github.com/hyprwm/hyprland-website/issues/24).

(Out of separate curiosity, however, what are these design change?)

zelfroster commented:
Dude, At least ask the devs before working on a big PR. Or look at old issues for a similar one.
There are already a few requests for using a Framework or SSG, but the owner is not open to ideas, Even I agree to his opinion, a simple website should not be made more complicated than it already is. Pure HTML + CSS + maybe some JS is more than enough for a simple website.

Also What are those commit messages? Looks like troll commits

zelfroster commented:
> Dude, At least ask the devs before working on a big PR. Or look at old issues for a similar one. There are already a few requests for using a Framework or SSG, but the owner is not open to ideas, Even I agree to his opinion, a simple website should not be made more complicated than it already is. Pure HTML + CSS + maybe some JS is more than enough for a simple website.
> 
> Also What are those commit messages? ~~Looks like troll commits~~ Improve them next time when you work, make them more clear.

Visual-Dawg commented:
Hey guys, I talked with Vax, its all good. 
Yes, the commits are a bit messy, but its actually a pretty new website and they should get squashed anyway.
I showed the progress on the Discord

NotAShelf commented:
P.S. you don't have to be smartasses

LeoDog896 commented:
> Hey guys, I talked with Vax, its all good. Yes, the commits are a bit messy, but its actually a pretty new website and they should get squashed anyway. I showed the progress on the Discord

Awesome! Are you planning to use GitHub actions for deployment, or are y'all continuing with the normal run on the server sort of situation?

zelfroster commented:
> Hey guys, I talked with Vax, its all good. Yes, the commits are a bit messy, but its actually a pretty new website and they should get squashed anyway. I showed the progress on the Discord

Didn't knew What's going on behind the scenes. I am actually surprised that Vaxry agreed for using anything else than simple stuff.

vaxerski commented:
I got ~~bribed~~ convinced to allow this

NotAShelf commented:
$20 is $20

vaxerski commented:
![image](https://github.com/hyprwm/hyprland-website/assets/43317083/5aaa4392-75fa-40cd-ac16-3afc007d3bb6)

footer is fucked, margin on the bottom?

discord section - to be removed imo (especially since it might disappear soon)

![image](https://github.com/hyprwm/hyprland-website/assets/43317083/5bf1f9b1-af19-4284-9def-71f09d1a41d9)

"other options"

![image](https://github.com/hyprwm/hyprland-website/assets/43317083/dd5220bd-b3fc-4fb9-aa9c-c655a0a0cdc5)

these animations here are annoying tbh, I am not a fan

Major note: the host I use cannot use pnpm. only npm.

Visual-Dawg commented:
Thank you for the feedback.
I will change those. 

The site is currently set to be build as a purely static site, so a host server is not nessecary. Something like GithubPages is already enough. 
I can change that though. And running the commands with their `npm` counterpart will have the same results in this case.

vaxerski commented:
ok then, no need to change shit if it's static

Visual-Dawg commented:
Added the changes. I can also take a look on how to use Github Pages with that, if you want

Vagahbond commented:
super cool looking :+1:

Visual-Dawg commented:
Note to self: Remove no-index when finished

Visual-Dawg commented:
This is the current status:
http://hyprtesty.surge.sh

I think its ready

vaxerski commented:
> I think its ready

and then man pushes like 30 commits

decide

vaxerski commented:
I told your ass to make this scroll and not redirect to wiki:
![image](https://github.com/hyprwm/hyprland-website/assets/43317083/ecf41382-6cd3-4cba-9b6b-cec3fe08c11e)

![image](https://github.com/hyprwm/hyprland-website/assets/43317083/5b3cb443-7579-4d97-aad1-6ceade2f9a89)

Check out

![image](https://github.com/hyprwm/hyprland-website/assets/43317083/336da7e1-5afc-4421-bbe1-f668d13ce858)

why does the HoF only contain the oldest rices? Why not random / winner of each competition?

![image](https://github.com/hyprwm/hyprland-website/assets/43317083/8b6c91bd-c8dc-43e0-b938-15d9bf80b39a)

why is this gibberish japanese in the back here

Visual-Dawg commented:
> and then man pushes like 30 commits
I received some more feedback from outfoxxed, which I wanted to include.

> I told your ass to make this scroll and not redirect to wiki:
The navbar will be visible on all pages, so it cant scroll down unless its the home page.
In general, I wanted this to be consistent. (All install buttons lead to the wiki)

> Check out
Ill fix this in a sec

> why does the HoF only contain the oldest rices? Why not random / winner of each competition?
Because their colors create the nicest perspective. Warmer colors are closer, coolers more far away. Here flick0s rice is perfect to be in front and then the last one has a lot of blue and blends nicely with the blue to create a sense of depth / atmosphere.
I played around with the new rices, but they are mostly black and it just doesnt look that nice.

> why is this gibberish japanese in the back here
It looks cool

vaxerski commented:
you can use this for japanese https://www.reddit.com/r/beemovie/comments/ge52xx/bee_movie_script_in_japanese/

Visual-Dawg commented:
For whatever reason I cannot update the site on surge.sh, but I did change the japanese text with a part of the linked post

vaxerski commented:
brilliant, it's the bee movie script in japanese

Visual-Dawg commented:
It also looks better now 👍 
If there isnt anything else, this can get merged

NotAShelf commented on src/routes/CommunitySlice.svelte:
```diff
@@ -0,0 +1,281 @@
+	<div class="group mt-16 flex flex-col items-center">
+		<a
+			class="discord p-4"
+			href="https://discord.com/invite/hQ9XvMUjjr"
+			target="_blank"
+			rel="noopener"
+			aria-label="Join us on Discord"
+		>
+			<DiscordIcon class="h-full w-full  " />
+		</a>
+
+		<a href="https://discord.com/invite/hQ9XvMUjjr">
+			<Button type="fancyOutline">Join us on Discord</Button>
+		</a>
```
I'm not familiar with Sveltekit, but in Next we usually have routes such as `domain.tld/discord` that we can then use inside the code to reduce redundancy and improve maintainability. If it's possible (and I'm guessing it is), I'll recommend poiting a single `hyprland.org/discord` to the discord invite and using that route in all discord links

NotAShelf commented on src/routes/FeaturesSlice.svelte:
```diff
@@ -0,0 +1,212 @@
+		</FeatureCard>
+		<FeatureCard class="" title="Dynamic tiling" color="purple">
+			<p class="max-w-[60ch]">
+				Automatic tiling which just works. Supports multiple fine-tuneable layouts.
```
Automatic tiling *that* just works.

NotAShelf commented on src/routes/hall_of_fame/+page.svelte:
```diff
@@ -0,0 +1,198 @@
+		/>
+	</Contest>
+
+	<Contest name="Summer" number={2} date="May 2023" --color="orange">
+		<FamedRice
+			name="Summer Gruv"
+			creator="end_4"
+			pretitel="#1"
+			dotfilesLink="https://github.com/end-4/dots-hyprland/tree/summer-gruv"
+			creatorProfilePicture="https://avatars.githubusercontent.com/u/97237370?s=24&v=4"
+			thumbnail="/imgs/ricing_competitions/2/end_4.webp"
+		/>
+		<FamedRice
+			name="Unnamed"
+			creator="Flafy"
+			pretitel="#2"
+			dotfilesLink="https://github.com/FlafyDev/flutter_background_bar"
+			creatorProfilePicture="https://avatars.githubusercontent.com/u/44374434?s=96&v=4"
+			thumbnail="/imgs/ricing_competitions/2/flafy.webp"
+		/>
+		<FamedRice
+			name="Day and Night"
+			creator="Mathisbuilder"
+			pretitel="#3"
+			dotfilesLink="https://github.com/MathisP75/summer-day-and-night"
+			creatorProfilePicture="https://avatars.githubusercontent.com/u/98901170?v=4"
+			thumbnail="/imgs/ricing_competitions/2/day-night.webp"
+		/>
+	</Contest>
```
I'd use a function that iterates over a json file to construct the FamedRice components here

NotAShelf commented on src/routes/hall_of_fame/+page.svelte:
```diff
@@ -0,0 +1,198 @@
+			creator="Flafy"
+			pretitel="#1"
+			dotfilesLink="https://github.com/FlafyDev/flutter_workspaces_2"
+			creatorProfilePicture="https://avatars.githubusercontent.com/u/25975326?v=4"
+			thumbnail="/imgs/ricing_competitions/1/flafy.webp"
+		/>
+		<FamedRice
+			name="Aurora"
+			creator="flick0"
+			pretitel="#2 (ex aequo)"
+			dotfilesLink="https://github.com/flick0/dotfiles/tree/aurora"
+			creatorProfilePicture="https://avatars.githubusercontent.com/u/77581181?s=48&v=4"
+			thumbnail="/imgs/ricing_competitions/1/flicko.webp"
+		/>
+		<FamedRice
+			name="Hyprland Winter"
+			creator="amadeus"
+			pretitel="#2 (ex aequo)"
+			dotfilesLink="https://github.com/AmadeusWM/hyprland-winter"
+			creatorProfilePicture="https://avatars.githubusercontent.com/u/63149896?s=48&v=4"
+			thumbnail="/imgs/ricing_competitions/1/amadeus.webp"
+		/>
+		<FamedRice
+			name="Unnamed"
+			creator="Lyasm"
+			pretitel="#3 (ex aequo)"
+			dotfilesLink="#"
+			creatorProfilePicture="https://avatars.githubusercontent.com/u/111616244?s=48&v=4"
+			thumbnail="/imgs/ricing_competitions/1/lyasm.webp"
+		/>
+		<FamedRice
+			name="Unnamed"
+			creator="lauroro"
+			pretitel="#3 (ex aequo)"
+			dotfilesLink="https://github.com/lauroro/hyprland-dotfiles"
+			creatorProfilePicture="https://avatars.githubusercontent.com/u/88981092?s=48&v=4"
+			thumbnail="/imgs/ricing_competitions/1/lauroro.webp"
+			class="pb-16"
+		/>
+	</Contest>
```
see above

Visual-Dawg replied:
Good points, Ill update the code

Visual-Dawg replied:
Code is updated :)

NotAShelf reviewed (commented):
LGTM

I've got some minor nitpicks, and other changes for improved maintainability - none of them are dealbreakers, but would be appreciated

vaxerski commented:
push it to hyprtesty and lmk

Visual-Dawg commented:
[ hyprtesty.surge.sh](http://hyprtesty.surge.sh/)
Aight, here you go

vaxerski commented:
all the easter egg tiles on the main page are laggy / buggy if you quickly hover off of them. No animations, they snap into place.

Visual-Dawg commented:
> all the easter egg tiles on the main page are laggy / buggy if you quickly hover off of them. No animations, they snap into place.

Damn, Firefox at it again. Ill see if I can work around that

NotAShelf commented:
also seeing minor frame drops on main page videos

Visual-Dawg commented:
Ahhhh, Ill probably the newish codecs. Much smiller videos, but takes more power to play.
Maybe Ill find a middle ground

Visual-Dawg commented:
Aight, Im using h264 again, which plays back smoothly and added a nice slow fade to the background images in the hero section.

I also re-encoded end_4's rice, so that the file can start playback without being fully downloaded and as I received feedback that the slide-on-hover in the plugins can be confusing (while I also received feedback that its weird if it does not slide) - I added a button which on click slides the video in.

Also, if there are some small optimizations here, I would prefer that the site still gets published/merged and that those then become issues, which I can than do in batches.

NotAShelf reviewed (approved):
looks good to go

vaxerski commented:
the tiles still snap back if I unhover while they are enlarged

Visual-Dawg commented:
@vaxerski Ah, I misunderstood you. I thought you were talking about the background-image.
Havnt spotted this one. 
Ill fix this one

Visual-Dawg commented:
Aight, I updated it

vaxerski reviewed (approved):
lgtm

vaxerski commented:
thanks!
